thin-replica-server/include/thin-replica-server/thin_replica_impl.hpp

```diff
@@ -253,11 +256,12 @@ class ThinReplicaImpl {
         LOG_INFO(logger_, "Subscription stream closed: " << error.what());
         break;
       }
-      metrics_->setLastSentBlockId(stream_type, getClientId(context), update.block_id);
+      metrics_.last_sent_block_id.Get().Set(update.block_id);
     }
     config_->subscriber_list.removeBuffer(live_updates);
     live_updates->removeAllUpdates();
-    metrics_->setSubscriberListSize(config_->subscriber_list.Size());
+    metrics_.subscriber_list_size.Get().Set(config_->subscriber_list.Size());
+    metrics_.updateAggregator();
```

Comment:
When do we need to update? Given that we should expect that this function never returns because it is an infinite stream of data, maybe we need to call it inside the while-loop too? Not sure, let's check `Metrics.hpp`.

Reply:
Thanks for catching this, updated.